catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -332,28 +344,88 @@ void registryInfoQuery() throws UnsupportedQueryException {
         SourceResponse identityMetacard = query(queryRequest);
         if (identityMetacard.getResults()
                 .size() > 0) {
-            remoteName = identityMetacard.getResults()
+            String metacardTitle = identityMetacard.getResults()
                     .get(0)
                     .getMetacard()
                     .getTitle();
+            setId(metacardTitle);
             registryId = RegistryUtility.getRegistryId(identityMetacard.getResults()
                     .get(0)
                     .getMetacard());
+            updateConfiguration(metacardTitle);
         }
-        updateConfiguration();
+
     }
 
-    private void updateConfiguration() {
+    private String getConnectionType(Bundle bundle, String pid) {
+        Locale locale = Locale.getDefault();
+        if (bundle != null) {
+            if (metaTypeService != null) {
+                MetaTypeInformation mti = metaTypeService.getMetaTypeInformation(bundle);
+                if (mti != null) {
+                    try {
+                        return mti.getObjectClassDefinition(pid, locale.toString())
+                                .getName();
+                    } catch (IllegalArgumentException e) {
+                        // MetaTypeProvider.getConnectionType might throw illegal
+                        // argument exception. So we must catch it here, otherwise the
+                        // other configurations will not be shown
+                        // See https://issues.apache.org/jira/browse/FELIX-2390
+                        // https://issues.apache.org/jira/browse/FELIX-3694
+                    }
+                }
+            }
+        }
+
+        // fallback to nothing found
+        return null;
+    }
+
+    public void setMetaTypeService(MetaTypeService metaTypeService) {
+        this.metaTypeService = metaTypeService;
+    }
+
+    private void updateConfiguration(String metacardTitle) {
+        if (metacardTitle == null) {
+            LOGGER.error("Unable to update registry configurations. No metacard title.");
+            return;
+        }
         String currentPid = getConfigurationPid();
         try {
             Configuration currentConfig = configAdmin.getConfiguration(currentPid);
             Dictionary<String, Object> currentProperties = currentConfig.getProperties();
-            currentProperties.put(REMOTE_NAME, remoteName);
+            URI uri;
+            String registryUrl = (String) currentProperties.get(REGISTRY_URL);
+            try {
+                uri = new URI(registryUrl);
+            } catch (URISyntaxException e) {
+                LOGGER.error("Unable to update registry configurations. Syntax error on registryUrl",
+                        e);
+                return;
+            }
+            String port = "";
+            if (uri.getPort() != -1) {
```

Comment:
java.net.URI.getPort() returns -1 if the port number is not defined in the URL that was given. So if the registryUrl has a port, ":" + uri.getPort() is appended to the nameToSet. Otherwise, the nameToSet string building continues normally but without the port appended. Should I set a string constant with a descriptive name to -1 and check uri.getPort() against it instead?